src/DmaRequestCore.bsv

```diff
@@ -22,78 +24,80 @@ module mkChunkComputer(ChunkCompute ifc);
 
     Reg#(DmaMemAddr) newChunkPtrReg <- mkReg(0);
     Reg#(DmaMemAddr) totalLenRemainReg <- mkReg(0);
-    Reg#(Bool) isSplittigReg <- mkReg(False);
+    Reg#(Bool) isSplittingReg <- mkReg(False);
 
     function Bool hasBoundary(DmaRequestFrame request);
-        let highIdx = (request.startAddr + request.length) >> BUS_BOUNDARY_WIDTH;
-        let lowIdx = request.startAddr >> BUS_BOUNDARY_WIDTH;
+        let highIdx = (request.startAddr + request.length - 1) >> valueOf(BUS_BOUNDARY_WIDTH);
+        let lowIdx = request.startAddr >> valueOf(BUS_BOUNDARY_WIDTH);
         return (highIdx > lowIdx);
     endfunction
 
     function DmaMemAddr getOffset(DmaRequestFrame request);
-        DmaMemAddr offset = zeroExtend(fromInteger(valueOf(BUS_BOUNDARY)) - pack(request.startAddr[BUS_BOUNDARY_WIDTH-1:0]));
-        return offset;
+        // 4096 - startAddr % 4096
```

Comment:
4096 is bus boundary, not tlp boundary. are you massing them up?

Reply:
Here, the calculated _offset_ is relative to _BUS_BOUNDARY_. The _offset_ will be sliced to _PCIE_TLP_BYES_ in the _getFirstChunkLen_ rule.

Reply:
Is there any need to split by BUS_BOUNDARY? If you split by PCIE_TLP_BYES , since PCIE_TLP_BYES <= BUS_BOUNDARY, any slice that aligned to PCIE_TLP_BYES won't cross BUS_BOUNDARY.

Reply:
You are right. I have modfied.